compiler/rustc_builtin_macros/src/deriving/smart_ptr.rs

```diff
@@ -143,31 +152,171 @@ pub fn expand_deriving_smart_ptr(
 
     // Find the `#[pointee]` parameter and add an `Unsize<__S>` bound to it.
     let mut impl_generics = generics.clone();
+    let pointee_ty_ident = generics.params[pointee_param_idx].ident;
+    let mut self_bounds;
     {
         let p = &mut impl_generics.params[pointee_param_idx];
+        self_bounds = p.bounds.clone();
         let arg = GenericArg::Type(s_ty.clone());
         let unsize = cx.path_all(span, true, path!(span, core::marker::Unsize), vec![arg]);
         p.bounds.push(cx.trait_bound(unsize, false));
         let mut attrs = thin_vec![];
         swap(&mut p.attrs, &mut attrs);
         p.attrs = attrs.into_iter().filter(|attr| !attr.has_name(sym::pointee)).collect();
     }
+    // We should not set default values to constant generic parameters
+    // and write out bounds that indirectly involves `#[pointee]`.
+    for (params, orig_params) in impl_generics.params[pointee_param_idx + 1..]
+        .iter_mut()
+        .zip(&generics.params[pointee_param_idx + 1..])
+    {
+        if let ast::GenericParamKind::Const { default, .. } = &mut params.kind {
+            *default = None;
+        }
```

Comment:
Also, doesn't this only remove the defaults from const generics that come *after* the `#[pointee]` param? Shouldn't we want to remove all of them?

Reply:
Applied. I should also remove the default type parameters.